lib/modules/datasource/rubygems/readme.md

```diff
@@ -42,7 +47,9 @@ Rubygems rate limits are easy to hit, so we need to be careful with the queries.
     delta_sync_error --> Empty: (3) Status other than 404 or 416
   ```
 
-- Then, more data is obtained from `https://rubygems.org/api/v1/versions/<package>.json` and `https://rubygems.org/api/v1/gems/<package>.json`
+- Then, more data is obtained from `https://rubygems.org/api/v1/versions/<package>.json` and `https://rubygems.org/api/v1/gems/<package>.json`.
+
+  Cache key is formed from the `packageName` and `version[]` from the previous layer, so that we reach these endpoints only when the list of versions has changed.
```

Comment:
Do you handle if there's a version mismatch between the two? i.e. don't treat it as a "finished" cache and retry again after some time?

Reply:
When the hash of `/versions` data doesn't match the `/api/v1/versions/<pkg>` response data, we return the `/versions` data (without `releaseTimestamp` and other metadata). It means querying `/api/v1/versions/<pkg>` until the consistency between two is achieved.

Reply:
So it keeps querying indefinitely?

Reply:
Okay, we have two options here:
1. Create additional code that protect us from this unfortunate case (which itself can be a source of bugs)
2. Leave as-is and observe how many log messages for inconsistent cache we get

Reply:
The downsides for both options are:
1. In similar conditions, instead of querying `/api/v1/versions/<pkg>` indefinitely, instead we'll return stale data indefinitely (can be logged and observed, though)
2. Creating PRs based only on the `version` fields will piss our users off time to time, so they'll come to report a bug. Not cool.

Reply:
If the API becomes correct then we want to pick it up. So this probably means retrying, such as daily. Maybe we could record the date when we first found the missing version and try up to a week?

Reply:
I don't think it's about days or weeks, for me it seems more like about minutes of inconsistency. If if there is something other than that, it looks like something we actually need to report to [rubygems.org](rubygems.org). But yes, I think we need some protective mechanism against us being rate limited.

Reply:
Just implemented mechanism of marking the cache as fallback and setting 1 more hour of TTL for it